SECURITY.md

```diff
@@ -110,6 +112,42 @@ time, we will work towards improved safe-by-default configuration, but due to ba
 compatibility and performance concerns, this will require following the breaking change deprecation
 policy.
 
+#### Core and extensions
+
+Anything in the Envoy core may be used in both untrusted and trusted deployments. As a consequence,
+it should be hardened with this model in mind.  Security issues related to core code will usually
+trigger the security release process as described in this document.
+
+Some [extensions](EXTENSION_POLICY.md) are also widely used and deserve the same treatment as core.
+Examples include HTTP connection manager, TCP proxy, buffer filter, router filter, CORS, dynamic
```

Comment:
Rather than explicitly enumerate extensions here, I think it would be better to have tag extensions based on the security hardening status. If we did this in the Bazel targets, we could even have Bazel query be able to do identify whether a given Envoy build has only hardened extensions. WDYT?

Reply:
Even if not auto-generated in this PR, I'd very much prefer to see a bullet list of filters using their well-known names.

Reply:
I just added a prototype for how I would think this would look. WDYT? Tagging @lizan as well for Bazel oversight.

Reply:
I think this is great, but I'd like to see the bullet point list of well-known names included in this document, or a separate file if that's easier to generated, instead of the comma separated list of examples that we have now.